### ISSUE TITLE
Packages to rebuild for openssl 3

### ISSUE THREAD

**Issue by pulkomandy** [haikuports/haikuports#10865]
List of recipes that use devel:libcrypto without a ">= 3" specifier (generated using `git grep "devel:libcrypto[^ p]*$"`):

- [ ] app-emulation/boxedwine/boxedwine-21.0.1.recipe (not compatible with openssl 3)
- [ ] dev-db/mysql/mysql-5.0.83.recipe (recipe is marked as broken on all architectures, untouched since being converted from bep file)
- [ ] dev-games/godot/godot-3.0.6.recipe (recipe marked as broken, made some "blind" changes)
- [x] dev-lang/php/php-7.4.30.recipe (not compatible with openssl 3)
- [ ] games-kids/tuxmath/tuxmath-1.7.2.recipe (Can't exec "autopoint": No such file or directory at /packages/autoconf-2.72-1/.self/data/autoconf/Autom4te/FileUtils.pm line 318.) segfaults
- [ ] net-im/kotatogram-desktop/kotatogram_desktop-1.4.9.recipe (CMake Error at cmake/options.cmake:88 (message): Unknown platform type)
- [x] www-client/otter-browser/otter_browser-1.0.03.recipe
- [x] www-servers/apache/apache-2.4.57.recipe
- [x] www-servers/lighttpd/lighttpd-1.4.53.recipe
- [x] webkit_gtk: fails to build on 32bit, does not find freetype

I am rebuilding them one by one after adding the >= 3 as needed, help welcome :)

**Comment by Begasus** [comment]
Already did the ones for Qt6 here, packages relying on Qt6 and using openssl should be done after Qt6 has been done (my 2 cents). Already got a "fixed" version for nexcloud-client here, will push when done.
@threedeyes I only saw one reference for openssl in the patches for Qt6, so I guess that should still be fine?

**Comment by jmairboeck** [comment]
I just built a new version of libpaper for my "project" of updating and unbundling texlive_core's dependencies. I'll take a look at R if it works with the new version and can also do openssl at the same time.

**Comment by lonemadmax** [comment]
Should the ones that declare devel dependency on libssl and not libcrypto be also rebuilt? Falkon is one such case reported in the forums. It has both libraries in requires, but only libssl is declared in build requires.

**Comment by pulkomandy** [comment]
Yes, I made the list in this ticket from devel:libcrypto, but devel:libssl should also be checked

**Comment by Begasus** [comment]
> Should the ones that declare devel dependency on libssl and not libcrypto be also rebuilt? Falkon is one such case reported in the forums. It has both libraries in requires, but only libssl is declared in build requires.

Already got an update to 23.08.5 for that, could push it, only 1 mentioned yesterday in the forum so I let him test the one I got here, seemed to be fine. Could push that.

**Comment by Begasus** [comment]
> Yes, I made the list in this ticket from devel:libcrypto, but devel:libssl should also be checked

From what I see "devel:libssl" is more in use as "devel:libcrypto"

**Comment by Begasus** [comment]
@threedeyes kotatogram_desktop build is broken (see below), is this still usefull as we already have a fully working Telegram port?

```
CMake Error in Telegram/lib_lottie/CMakeLists.txt:
  Imported target "external_lz4" includes non-existent path

    "/usr/local/develop/headers"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

**Comment by Begasus** [comment]
OK, think I've gone as far as I can on cleaning the list, things not compatible I haven't checked, some obsolete, and with rebuild for tuxmath I get a weird segfault (didn't investigate further, source hasn't been updated in a long time).

**Comment by pulkomandy** [comment]
I forgot to note it here, for php I updated php8 without problems, I think we can drop the php 7 recipe.

**Comment by Begasus** [comment]
Removed php7 from the system without problems, should be ok to drop.

Did a quick search on the repository here, still quite a bit there, probably some don't need revbump, but a few probably will have to, I guess we can tackle the ones on a version bump.

**Comment by Begasus** [comment]
> `minidlna` can be removed from this list. It doesn't requires `libssl`/`libcrypto` at all (got it to build and run after disabling `kqueue` detection).

Removed from the list.

**Comment by jmairboeck** [comment]
#14618 is a duplicate of this, but that list is more complete than this, so I'm closing this one instead.
